app/main.py

```diff
@@ -1,3 +1,83 @@
-def shop_trip():
-    # write your code here
-    pass
+from math import sqrt
+
+from app.car import FUEL_PRICE
+from app.customer import customers_
+from app.shop import shops_
+
+
+def shop_trip() -> None:
+    customers = customers_
+    shops = shops_
+    fuel_price = FUEL_PRICE
+
+    for customer in customers:
+        cheapest_shop = []
+        print(f"{customer.name} has {customer.money} dollars")
+        for shop in shops:
+            total_price = 0
+            for (_, value_need_customer), (_, shop_item_price) in zip(
```

Comment:
Change variables names, they are not really descriptive
I don't really understand what value_need_customer means.
You can also change shop_item_price to product_price, since it means the same and we use `product` naming here